pages/api/id/[form]/bearer.ts

```diff
@@ -0,0 +1,45 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { getSession } from "next-auth/client";
+import queryManager from "../../../../lib/integration/queryManager";
+import isRequestAllowed from "../../../../lib/middleware/httpRequestAllowed";
+
+const allowedMethods = ["GET"];
+
+export const retrieve = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+  try {
+    const session = await getSession({ req });
+    // Only allow a valid user session
+    if (!session || session === null || req.method !== "GET")
+      return res.status(403).json({ error: "Access Denied" });
+    await getToken(res, req.query.form as string);
+  } catch (err) {
+    res.status(500).json({ error: `Error on Server Side ${err}` });
+  }
+};
+/**
+ * Will return a bearer token if there is one associated with a given formID
+ * It might return a Not found (404) if token doest exist ( undefined or null)
+ * otherwise 400 as status code.
+ * @param formID A form id
+ */
+async function getToken(res: NextApiResponse, formID: string) {
```

Comment:
probably best to separate this out in a separate file in the lib folder so it can be utilized elsewhere. This can even be made into a middleware where the resulting handler gets injected with a token. In any case in the interest of having this PR done we can skip this for now but just a note in the future